packages/flutter/lib/src/material/typography.dart

```diff
@@ -139,24 +172,62 @@ class TextTheme extends Diagnosticable {
   /// the value of [TextStyle.inherit] flag. For more details, see the
   /// documentation on [TextStyle.merge] and [TextStyle.inherit].
   ///
+  /// If this theme, or the `other` theme has members that are null, then the
+  /// non-null one (if any) is used. If the `other` theme is itself null, then
+  /// this [TextTheme] is returned unchanged. If values in both are set, then
+  /// the values are merged using [TextStyle.merge].
+  ///
   /// This is particularly useful if one [TextTheme] defines one set of
-  /// properties and another defines a different set, e.g. having colors defined
-  /// in one text theme and font sizes in another.
+  /// properties and another defines a different set, e.g. having colors
+  /// defined in one text theme and font sizes in another, or when one
+  /// [TextTheme] has only some fields defined, and you want to define the rest
+  /// by merging it with a default theme.
+  ///
+  /// ## Sample code
+  ///
+  /// ```dart
+  /// /// A Widget that sets the ambient theme's title text color for its
+  /// /// descendants, while leaving other ambient theme attributes alone.
+  /// class TitleColorTheme extends StatelessWidget {
+  ///   TitleColorTheme({Key key, this.child, this.titleColor}) : super(key: key);
+  /// 
+  ///   final Color titleColor;
+  ///   final Widget child;
+  /// 
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     ThemeData theme = Theme.of(context);
+  ///     // This partialTheme is incomplete: just replacing theme.textTheme with
+  ///     // partialTheme wouldn't do the right thing, so merge it with the existing theme.
+  ///     TextTheme partialTheme = new TextTheme(title: new TextStyle(color: titleColor));
+  ///     theme = theme.copyWith(textTheme: theme.textTheme.merge(partialTheme));
+  ///     return new Theme(data: theme, child: child);
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// Using [copyWith] instead can make more sense in some situations.
```

Comment:
Same comment as the one about "Using [merge]"

Reply:
Done.